Add type narrowing in templates

artemdemo opened the pull request:
In this PR I'm adding types narrowing with SDK types
https://wix.atlassian.net/browse/FEDINF-13511

fantua commented on chart-widget/template/src/site/widgets/custom-elements/chart/panel.tsx:
```diff
@@ -1,14 +1,13 @@
 
-function Panel() {
-  const { setProp, getProp } = useWixModules(widget);
+export default () => {
```
```suggestion
const Panel: FC = () => {
```
Consider `FC` type usage to be consistent with `cli-app` template

AdirAmsalem replied:
Thanks, done

fantua commented on chart-widget/template/src/site/widgets/custom-elements/chart/panel/slice.tsx:
```diff
@@ -18,7 +17,7 @@ interface Props {
 }
 
 export default function Slice({ title, item, onChange }: Props) {
```
same here and in other files

artemdemo committed:
Add type narrowing in templates

In this PR I'm adding types narrowing with SDK types
https://wix.atlassian.net/browse/FEDINF-13511
